firmware/test_protect_payload_firmware/main.rs

```diff
@@ -0,0 +1,33 @@
+#![no_std]
```

Comment:
It would be good to have a few doc comments here, those tests are going to be non-trivial, so describing what is happening (e.g. once the workflow you want to test is clearer we should describe it in a doc comment). This is true for the other firmware/payloads too.

Reply:
classique comment :)